### ISSUE TITLE
Escape chars with sed

### ISSUE THREAD

**Issue by FlyingWombat** [mikebattista/PowerShell-WSL-Interop#18]
**Describe the bug**
Backslash in escape chars are converted to forward-slash, which makes regex in sed, grep, etc. not work.

**To Reproduce**
```
PS > echo "asdf" | sed 's/d/\n/'
asnf
```
```
PS > cat Env:\Path | sed 's/;/\n/g'
sed: -e expression #1, char 2: unterminated 's' command
/bin/bash: /n/g: No such file or directory
```
**Expected behavior**
```
PS > echo "asdf" | sed 's/d/\n/'
as
f
```
```
PS > cat Env:\Path | sed 's/;/\n/g'
C:\Windows\system32
C:\Windows
...
```

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Powershell 7.0 preview4
 - git master 257eec1

**Comment by mikebattista** [comment]
Looks like the problem isn't the slashes but the semicolon. If you escape that (\;) it will work just fine.

I generally use the Format-WslArgument helper to escape things like this under the hood so you can provide arguments naturally and not have to think about how they are translated to WSL commands, but I can't do it for everything without breaking legitimate use cases for some special characters.

The fix here would be to escape ';' whenever they occur in any arguments, but I'll need to think about if that will have any negative consequences.

**Comment by mikebattista** [comment]
For your first use case, if you escape the \\ in the \\n, so \\\\n, it will work as expected.

The common problem here is the single quotes you type in PowerShell are used by PowerShell itself to not parse the content. Those single quotes are not passed to WSL as you might expect, so the commands are behaving like the quoted argument was passed on the Linux command line without quotes. So the ; is interpreted as a command separator, and \n is treated as an escape sequence for 'n'.

Another option would be to insert quotes within the single quoted string, so '''s/d/\\n/''' for example which will result in single quotes being sent down to WSL as part of the argument.

Obviously this is less than ideal which is why I try to handle this under the hood with Format-WslArgument. I could add quotes under the hood but that has some negative consequences.

I'll need to think about if there's a smart way to handle this without breaking legitimate use cases.

**Comment by mikebattista** [comment]
A first attempt at fixing this is available at [https://github.com/mikebattista/PowerShell-WSL-Interop/tree/mikebattista/special-characters](https://github.com/mikebattista/PowerShell-WSL-Interop/tree/mikebattista/special-characters) if you want to try it out.

It fixes your scenarios but needs some more testing to ensure it hasn't broken others.

**Comment by mikebattista** [comment]
Fixed by 131b5d544b6b6e74e008d69123a9e43d961fedef.

Updated the module on PSGallery.
